data-prepper-core/src/main/java/org/opensearch/dataprepper/peerforwarder/client/PeerForwarderClient.java

```diff
@@ -48,40 +54,57 @@ public AggregatedHttpResponse serializeRecordsAndSendHttpRequest(
             final String pluginId) {
         // TODO: decide the default values of peer forwarder configuration and move the PeerClientPool to constructor
         peerClientPool = peerForwarderClientFactory.setPeerClientPool();
-        List<WireEvent> wireEventList = new ArrayList<>();
 
-        for (Record<Event> record : records) {
-            final Event event = record.getData();
-            wireEventList.add(new WireEvent(event.getMetadata().getEventType(), event.toJsonString()));
+        final WebClient client = peerClientPool.getClient(ipAddress);
+
+        final String serializedJsonString = getSerializedJsonString(records, pluginId);
+        if (serializedJsonString == null) {
+            return AggregatedHttpResponse.of(HttpStatus.BAD_REQUEST);
         }
 
+        final CompletableFuture<AggregatedHttpResponse> aggregatedHttpResponseCompletableFuture = processHttpRequest(client, serializedJsonString);
+
+        return getAggregatedHttpResponse(aggregatedHttpResponseCompletableFuture);
+    }
+
+    private String getSerializedJsonString(final Collection<Record<Event>> records, final String pluginId) {
```

Comment:
I would recommend returning Optional. Then L61-63 could be compactified into a single expression.